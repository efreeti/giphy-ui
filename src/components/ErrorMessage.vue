<template>
  <div>
    <div>
      <md-icon class="md-size-5x md-accent">error</md-icon>
    </div>
    <div>
      <span class="error-message-text md-body-2">
        <slot />
      </span>
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

@Component
export default class ErrorMessage extends Vue {
}
</script>

<style lang="scss" scoped>
@import "~vue-material/dist/theme/engine";

.error-message-text {
  color: md-get-palette-color(red, A200);
}
</style>
