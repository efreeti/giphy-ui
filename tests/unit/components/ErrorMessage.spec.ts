import { mount } from '@vue/test-utils';

import '../config';

import ErrorMessage from '@/components/ErrorMessage.vue';

describe('ErrorMessage', () => {
  it('should render correctly', () => {
    expect(mount(ErrorMessage, {
      slots: {
        default: '<span>error message</span>',
      },
    }))
        .toMatchSnapshot();
  });
});
