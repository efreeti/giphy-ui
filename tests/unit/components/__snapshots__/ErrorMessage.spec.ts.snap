// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage should render correctly 1`] = `
<div>
  <div><i class="md-icon md-icon-font md-size-5x md-accent md-theme-default">error</i></div>
  <div><span class="error-message-text md-body-2"><span>error message</span></span></div>
</div>
`;
